datalad/core/distributed/clone.py

```diff
@@ -327,7 +327,8 @@ def clone_dataset(
         description=None,
         result_props=None,
         cfg=None,
-        checkout_gitsha=None):
+        checkout_gitsha=None,
+        misbuilt_url=None):
```

Comment:
although it is just an internal (albeit without `_` prefix) function, so we could stick anything into its signature, this `misbuilt_url` "memo"  or "info" of a kind makes me wonder, and please pardon if you already explored that - I failed to find in current comments, why just not to yield some "error" record with some annotation which higher level logic could then collect/filter and report on if so desired?

[*] btw , docstring for it isn't explicit that clone_dataset just adds to it and nohow uses it otherwise

Reply:
I echo the notion of "rather use the results" instead of in-place modification of some internal construct. But I am not in favor of "some error result", if that means changing the *number* of results reported (see additional comment of mine in a second).

Reply:
Sorry, I lost track of this PR. Can you help me understand how I could implement what you mean with:

>  why just not to yield some "error" record with some annotation which higher level logic could then collect/filter and report on if so desired

Is it that I should attempt cloning with the URL where the key error would originate, and add something to the result record that hints at the missing property in the template?

Reply:
I was thinking about smth like
```
                yield get_status_dict(
                    status='warning',
                    message="URL was misbuilt: {some_details}"
                    **result_props)
                continue
```
but given @mih 's comments "above and [below](https://github.com/datalad/datalad/pull/5644#issuecomment-860576337)  " on an expectation that this particular call should have only a singular status record, then indeed could/should go into some other extra field within the resultant record.

but then the question is also -- why should we try misbuilt urls at all?  so this spot doesn't seem to be the right one to handle it IMHO, should be done upstairs (thus not even calling `clone_dataset` if we know that URL is not good) and then just resultant record extended with that extra information.  Or am I missing some use case?

Reply:
I don't think you have missed a usecase, I think I just lacked insight on how to do it more cleverly. :) But how can I include one or more bad url(s) in the result record "upstairs" - isn't the final result rendered only in clone? How can I pass it into the record without passing it to clone?
If it would speed things up, I can also force push my earlier commits away, and leave it to someone else to commit a better approach, result rendering is still one of the many mysteries to me...